test/integration/operator_policy_error_test.go

```diff
@@ -0,0 +1,157 @@
+				"-n", userNamespace, "--type=json", "-p", `[{
+				"op":"replace", 
+				"path":"/spec/policy-templates/0/objectDefinition/spec/subscription/source",
+				"value":"redhat-operators-invalid"
+			}]`)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It(operatorPolicyName+" should be NonCompliant due to incorrect source", func() {
+			By("Checking if the status of the operator policy is NonCompliant")
+			Eventually(
+				common.GetComplianceState(policyName),
+				defaultTimeoutSeconds,
+				1,
+			).Should(Equal(policiesv1.NonCompliant))
+			Eventually(
+				common.GetOpPolicyCompMsg(operatorPolicyName),
+				defaultTimeoutSeconds,
+				1,
+			).Should(ContainSubstring("CatalogSource 'redhat-operators-invalid' was not found"))
+		})
+
+		It(operatorPolicyName+" should be patched with correct policy", func() {
+			_, err := common.OcHub(
+				"apply", "-f", correctPolicyPath, "-n", userNamespace,
+			)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It(operatorPolicyName+" should be patched with incorrect installPlanApproval", func() {
+			_, err := common.OcHub(
+				"patch", "policies.policy.open-cluster-management.io", policyName,
+				"-n", userNamespace, "--type=json", "-p", `[{
+				"op":"replace", 
+				"path":"/spec/policy-templates/0/objectDefinition/spec/subscription/installPlanApproval",
+				"value":"Invalid"
+			}]`)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It(operatorPolicyName+" should be NonCompliant", func() {
+			By("Checking if the status of the operator policy is NonCompliant")
+			Eventually(
+				common.GetComplianceState(policyName),
+				defaultTimeoutSeconds,
+				1,
+			).Should(Equal(policiesv1.NonCompliant))
+			Eventually(
+				common.GetOpPolicyCompMsg(operatorPolicyName),
+				defaultTimeoutSeconds,
+				1,
+			).Should(ContainSubstring("the policy spec.subscription.installPlanApproval ('Invalid') is invalid"))
+		})
+
+		It(operatorPolicyName+" should be patched with correct policy", func() {
+			_, err := common.OcHub(
+				"apply", "-f", correctPolicyPath, "-n", userNamespace,
+			)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It(operatorPolicyName+" should have namespace removed", func() {
+			_, err := common.OcHub(
+				"patch", "policies.policy.open-cluster-management.io", policyName,
+				"-n", userNamespace, "--type=json", "-p", `[{
+				"op":"remove", 
+				"path":"/spec/policy-templates/0/objectDefinition/spec/subscription/namespace",
+			}]`)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It(operatorPolicyName+" should be NonCompliant", func() {
+			By("Checking if the status of the operator policy is NonCompliant")
+			Eventually(
+				common.GetComplianceState(policyName),
+				defaultTimeoutSeconds,
+				1,
+			).Should(Equal(policiesv1.NonCompliant))
+			Eventually(
+				common.GetOpPolicyCompMsg(operatorPolicyName),
+				defaultTimeoutSeconds,
+				1,
+			).Should(ContainSubstring("namespace is required in spec.subscription"))
+		})
+	})
```

Comment:
Can we add error for `startingCSV` not existing version?

Reply:
These would have to be added into the actual controller code AFAIK, @JustinKuli any thoughts on this?

Reply:
If I had done https://issues.redhat.com/browse/ACM-9896 correctly the first time, this would have been obvious sooner, but `namespace` shouldn't be required in the OperatorPolicy, when it's deployed on OpenShift like it is in these tests. So this test case will need to be updated (possibly just removed).